### DISCUSSION TITLE
关于训练集训练次数以及pytorch版本问题

### DISCUSSION THREAD

**Discussion by lixu010701** [PengfeiM/Fatigue-Driven-Detection-Based-on-CNN#45] [General]
各位好，我在运行您的代码时遇到了两个问题，在csnd，百度等网站搜索了一天还没有解决，所以想请教各位一下，如果各位有时间，请赐教，谢谢，我的疑问如下：

1、运行 Train.py 训练时，需要训练到什么程度，我今天跑了一下午的程序，运行了十几个epoch，不知道要到什么时候停止，是需要自己设置训练次数吗，还是需要一直等程序自己运行结束。
2、pytorch版本的问题，我的版本是1.11.0，运行eval.py时，会出现在GitHub项目主页发布的更新说明一样，出现以下这种错误：
针对近来很多同学反映的在新版本pytorch下程序报错： "RuntimeError: Legacy autograd function with non-static forward method is deprecated. Please use new-style autograd function with static forward method. (Example: https://pytorch.org/docs/stable/autograd.html#torch.autograd.Function)" 。 做了修改了detection、Test、xxx_detection等部分代码，可适应新版本pytorch。另外旧版本pytor请选择old-pytorch分支。
目前GitHub上面的该项目，楼主应该已经做了修改了detection、Test、xxx_detection等部分代码，可在我运行的时候还是会出现以上错误，我目前仍然解决不了。

以上两个问题想请教各位一下，还烦请各位在空闲时间能给予帮助，感激不尽！

**Comment by XRA777** [comment]
关于问题1，看代码，我看到的是有两个终止条件，1.iter >= Config.max_iter，程序默认是120000；2.或者epoch大于1000。纯小白，从程序里就看出这些。。我也是要训练好久
